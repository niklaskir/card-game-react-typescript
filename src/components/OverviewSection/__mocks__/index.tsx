import React from 'react';

const OverviewSection = () => <div id="overview_section" />;

export default OverviewSection;
