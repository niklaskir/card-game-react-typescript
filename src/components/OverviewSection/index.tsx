import React from 'react';
import { useRecoilValue, useSetRecoilState } from 'recoil';
import {
  sortedUserList as sortedUserListState,
  userSelected as userSelectedState,
  UserType,
} from '../../recoil/states/userList';

const OverviewSection = () => {
  const sortedUserList = useRecoilValue(sortedUserListState);
  const setUserSelected = useSetRecoilState(userSelectedState);

  const handleSelect = (key: number) => {
    setUserSelected(sortedUserList[key]);
  };

  return (
    <div className="col-span-1 lg:col-span-2">
      <div className="font-bold text-xl mb-2">Overview</div>
      <div className="grid grid-cols-1 lg:grid-cols-3 gap-4">
        {sortedUserList.map((user: UserType, key: number) => (
          <div
            key={key}
            className="col-span-1 cursor-pointer"
            onClick={() => handleSelect(key)}
          >
            <div className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100">
              <div className="px-6 py-4">
                <ul className="list-disc">
                  <li className="text-gray-700 text-base">{user.realName}</li>
                  <li className="text-gray-700 text-base">{user.playerName}</li>
                  <li className="text-gray-700 text-base">{user.asset}</li>
                </ul>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default OverviewSection;
