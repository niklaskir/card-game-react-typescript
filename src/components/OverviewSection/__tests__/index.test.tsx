import users from '../../../data.json';
import React from 'react';
import Recoil from 'recoil';
import renderer from 'react-test-renderer';
import OverviewSection from '../index';

jest.mock('recoil', () => ({
  atom: jest.fn(),
  useRecoilState: jest.fn(),
  useSetRecoilState: jest.fn(),
}));

describe('OverviewSection', () => {
  beforeEach(() => {
    Recoil.useRecoilState = jest.fn().mockReturnValueOnce([users, jest.fn()]);
  });

  it('should render correctly', () => {
    const overviewSection = renderer.create(<OverviewSection />).toJSON();

    expect(overviewSection).toMatchSnapshot();
  });

  it('should call setUserSelected with correct params', () => {
    const setUserSelectedSpy = jest.fn();

    Recoil.useSetRecoilState = jest
      .fn()
      .mockReturnValueOnce(setUserSelectedSpy);

    const overviewSection = renderer.create(<OverviewSection />);

    overviewSection.root
      .findAllByProps({
        className: 'col-span-1 cursor-pointer',
      })[0]
      .props.onClick();

    expect(setUserSelectedSpy).toHaveBeenCalledWith(users[0]);
  });
});
