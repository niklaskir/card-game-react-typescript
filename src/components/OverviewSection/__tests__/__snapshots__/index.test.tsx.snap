// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewSection should render correctly 1`] = `
<div
  className="col-span-1 lg:col-span-2"
>
  <div
    className="font-bold text-xl mb-2"
  >
    Overview
  </div>
  <div
    className="grid grid-cols-1 lg:grid-cols-3 gap-4"
  >
    <div
      className="col-span-1 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100"
      >
        <div
          className="px-6 py-4"
        >
          <ul
            className="list-disc"
          >
            <li
              className="text-gray-700 text-base"
            >
              Brianna Forbes
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Dreamlurk The Unstoppable
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Foghammer Lead
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-span-1 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100"
      >
        <div
          className="px-6 py-4"
        >
          <ul
            className="list-disc"
          >
            <li
              className="text-gray-700 text-base"
            >
              Darcy Candice Ball
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Crystaldash
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Secret Glowquake Gold
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-span-1 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100"
      >
        <div
          className="px-6 py-4"
        >
          <ul
            className="list-disc"
          >
            <li
              className="text-gray-700 text-base"
            >
              Hillary Gibbs
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Speedsoul
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Shifting Rainshadow Iron
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-span-1 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100"
      >
        <div
          className="px-6 py-4"
        >
          <ul
            className="list-disc"
          >
            <li
              className="text-gray-700 text-base"
            >
              Elva Becky Hammond
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Seekvenom The Mystic
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Valkyries' Opal Adamant
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-span-1 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100"
      >
        <div
          className="px-6 py-4"
        >
          <ul
            className="list-disc"
          >
            <li
              className="text-gray-700 text-base"
            >
              Enid Rose
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Coincurse The Ghoul
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Jewelevil Bronze Of Goddesses
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-span-1 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="rounded overflow-hidden shadow-lg bg-slate-100 hover:bg-teal-100"
      >
        <div
          className="px-6 py-4"
        >
          <ul
            className="list-disc"
          >
            <li
              className="text-gray-700 text-base"
            >
              Esmeralda Carrillo
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Skulldart
            </li>
            <li
              className="text-gray-700 text-base"
            >
              Yellow Orichalcum Of Paladins
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
